lib/classes/Variables.test.js

```diff
@@ -1129,6 +1129,31 @@ module.exports = {
         });
     });
 
+    it('should NOT warn if an SSM parameter does not exist and no fallback given', () => {
+      const options = {
+        stage: 'prod',
+        region: 'us-east-1',
+      };
+      serverless.variables.options = options;
+      const awsProvider = new AwsProvider(serverless, options);
+      const param = '/some/path/to/invalidparam';
+      const property = `\${ssm:${param}}`;
+      const error = new serverless.classes.Error(`Parameter ${param} not found.`, 400);
+      const requestStub = sinon.stub(awsProvider, 'request', () => BbPromise.reject(error));
+      const warnIfNotFoundSpy = sinon.spy(serverless.variables, 'warnIfNotFound');
+
+      return serverless.variables.populateProperty(property)
+        .should.be.rejectedWith(serverless.classes.Error)
+        .then(() => {
+          expect(requestStub.callCount).to.equal(1);
+          expect(warnIfNotFoundSpy.callCount).to.equal(0);
+        })
+        .finally(() => {
+          requestStub.restore();
```

Comment:
Restores like this should be moved in a `beforeEach` / `afterEach` block so that the test suite takes care of the cleanup.